projects/storefrontlib/src/lib/product/components/product-list/container/product-list.component.ts

```diff
@@ -41,7 +45,7 @@ export class ProductListComponent implements OnChanges, OnInit {
 
   ngOnChanges() {
```

Comment:
please pass `SimpleChanges` to it.
`ngOnChanges(changes: SimpleChanges) {...}`.

In this `ngOnChanges`, `if only changes['gridMode']`, we will not do search.